### DISCUSSION TITLE
Streaks

### DISCUSSION THREAD

**Discussion by benedictweis** [SE-TINF22B2/G5-DuoGradus#5] [Anforderungen]
Die Anwendung soll dem Benutzer anzeigen seit wie vielen Tagen in folge er jetzt seine Schrittziele (#4)  erfüllt. Sie soll den Benutzer zusätzlich zu ausgewählten Zeiten daran Erinnern seinen Streak am laufen zu halten. Mögliche Erinnerungen wären:

- "Hey! Du hast heute noch keine 6000 Schritte gemacht. Geh raus und sammle ein paar."
- "Hallo! Du sitzt heute schon seit 2 Stunden, wie wärs mit ein paar Schritten?"
- "Achtung! Es ist schon 20:00Uhr, willst du heute etwa deinen Streak verlieren?"

Diese Erinnerungen sollen ähnlich aggressiv formuliert sein wie die in Duolingo.

**Comment by wwwLuis** [comment]
als  Bonus (wenn man noch zeit über hat) fände ich es eigentlich witzig, wenn man die Benachrichtigungen dynamisch von einer AI mit einem "etwas fiesen" prompt template erzeugen lassen würde

**Reply by benedictweis** [reply]
gute Idee, verursacht allerdings eventuell kosten, diese würden dann durch @henrybrink 's Kreditkarte gedeckt werden 😊

**Reply by henrybrink** [reply]
Glücklicherweise besitzt @henrybrink keine Kreditkarte und wird auch nicht nochmal für Kosten aufkommen, die Benedict bei OpenAI verursacht.

**Comment by henrybrink** [comment]
Die Streaks sollen zudem vom "Rang" des Benutzers abhängig sein. Ein Nutzer kann unter bestimmten Bedingungen "aufsteigen", hat dadurch dann aber auch ein höheres Schrittziel.

Beispiele für Ränge:

- Novize / Starter / Anfänger
- Schrittsammler
- Jogging-Meister
- Wandervogel
